addons/mail/models/mail_thread.py

```diff
@@ -4394,13 +4394,18 @@ def _message_followers_to_store(
                 ("partner_id.active", "=", True),
             ]
             domain = expression.AND([domain, subtype_domain])
-        if after:
-            domain = expression.AND([domain, [("id", ">", after)]])
+        followers_count = self.env["mail.followers"].search_count(domain, limit=151)
+        if followers_count <= 150:
+            followers = self.env["mail.followers"].search(domain, offset=offset, limit=limit, order="name ASC")
+        else:
+            if after:
+                domain = expression.AND([domain, [("id", ">", after)]])
+            followers = self.env["mail.followers"].search(domain, offset=offset, limit=limit, order="id ASC")
```

Comment:
I would return a sorted list in all cases, the threshold of `150` has no motivation

Reply:
Don't you think it will be computationally more taxing to sort a large number of followers alphabetically than based on their id? We can do that until the number of followers is small, once it reaches certain threshold ( here 150 ) it seems more feasible to sort them according to their id only. That was the initial thought behind implementing it this way

Reply:
@akap-odoo Can you try with 30k followers whether this is perceivably slow. We have actual cases of such amount of followers, so that's a good benchmark to tell whether this is an issue or not.